restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -1102,6 +1102,23 @@ private String getParentPathFromRow(Row row) {
     return s.toString();
   }
 
+  private boolean pathsMatch(String path1, String path2) {
+    String[] parts1 = path1.split("\\.");
```

Comment:
If this is on the hot-path, maybe it will be more efficient to have the compiler Pattern and reused it here?

Reply:
I made a Pattern for this and actually changed a whole bunch of `split` calls to use it instead :) thanks for pointing this out!